src/Security/Authentication/Negotiate/src/LdapOptions.cs

```diff
@@ -40,8 +45,25 @@ public class LdapConnectionOptions
         public bool ResolveNestedGroups { get; set; } = true;
 
         /// <summary>
-        /// Additional configuration on the created LdapConnection.
+        /// The LdapConnection to be used to retrieve role claims. If no explicit
+        /// connection is provided, an LDAP connection will be automatically created
+        /// based on the Domain, MachineAccountName and MachineAccountPassword options.
         /// </summary>
-        public Action<LdapConnection> ConfigureLdapConnection { get; set; }
+        public LdapConnection LdapConnection { get; set; }
+
+        public void Validate()
+        {
+            if (EnableLdapRoleClaimResolution)
+            {
+                if (string.IsNullOrEmpty(Domain))
```

Comment:
Is Domain required if you've specified the LdapConnection directly?

Reply:
Eh... probably not. we should just use the connection.